src/plugins/intel_cpu/src/nodes/reorder.cpp

```diff
@@ -262,17 +254,16 @@ void Reorder::createReorderPrimitive(const dnnl::memory::desc& srcDesc,
     // perform such conversion if the source tensor can be reshaped to the destination rank. This is
     // useful in situations when rank in IR does not much rank that is required by the oneDNN primitive,
     // but the input tensor can be reshaped (e.g. weights for grouped convolutions, biases etc.)
-    if (src_blocked->getDesc().hasLayoutType(LayoutType::ncsp) &&
-        src_blocked->GetShape().getRank() != dst_blocked->GetShape().getRank()) {
-        const auto newDims = dst_blocked->getStaticDims();
+    auto &srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
+    auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+    if (srcMemPtr->getDesc().hasLayoutType(LayoutType::ncsp) &&
+        srcMemPtr->GetShape().getRank() != dstMemPtr->GetShape().getRank()) {
```

Comment:
If we use `DnnlMemoryDescPtr` we can avoid these calls to the memory objects, all those data (rank and layout type) may be retrieved from descriptors.

Reply:
fixed, thank you Maksim.